### ISSUE TITLE
Bluetooth issues connecting to Wayteq x995

### ISSUE THREAD

**Issue by tchewitt** [fhorinek/SkyDrop#202]
Hi,

I'm having bluetooth pairing issues between Skydrop and Wayteq x995 (android 4.2.2 sat nav/tablet). Skydrop is new version running 2650. The x995 can see the Skydrop via bluetooth and attempts to connect. I know that these two devices have been connected by other users.

During pairing, sometimes two Skydrops are discovered with different logos; one called "Skydrop", the other "Skydrop pan1026". Other times only one device is discovered.

Two outcomes occur:

One of the two discovered skydrops pairs as normal, but after ~20 seconds the x995 reports "unable to pair due to incorrect pin or password". Pin or password cannot be changed. During the 20 second period where the devices appear connected xcsoar data monitor reports "Device Occupied". No data is visible in xcsoar.

or

Only one device is discovered. This device cannot be paired.

Thanks

Tom

**Comment by piman-34** [comment]
I have basically the same issue. Having got a BT adaptor to work in a Kobo mini I learned a bit about this but have just bought a no-brand Android tablet for XCSoar with logging on live track 24 and so on.

I ensured it had BT4.1 capability
Info is Android 4.4.2 based on kernel 3.4.67

Error is the same: if only Skydrop Pan 1026 is seen, pairing appears to work on the Skydrop but errors on the tablet after x seconds with pin error. Changing Tablet visibility sometimes reveals a second Skydrop appears with a joystick logo - this pairs with the tablet but doesn't connect on the device. Will collect some logs later but recall on the latest fw had trouble seeing BT activity in the logs.
![image](https://cloud.githubusercontent.com/assets/18732908/17525527/f40ea674-5e5a-11e6-9cda-9cd0cfba768d.jpeg)
![image](https://cloud.githubusercontent.com/assets/18732908/17525528/f42ccc44-5e5a-11e6-874d-575c2fce5b8e.jpeg)
![image](https://cloud.githubusercontent.com/assets/18732908/17525529/f42dd88c-5e5a-11e6-98fa-42088cf9a248.jpeg)

**Comment by tchewitt** [comment]
Yes, this appears identical to my issue. Thanks for improving the issue with screens etc. I was going to add some later tonight.

**Comment by fhorinek** [comment]
Hi guys, the bluetooth module pan1026 have dual mode operation.
SkyDrop SPP (clasic 2.1 version) and SkyDrop BLE (low energy version).
Only SPP will need (and support) pairing.

this (unofficial) release have most recent bluetooth drivers for pan1026 and also bluetooth debug.
[SKYDROP.FW.zip](https://github.com/fhorinek/SkyDrop/files/409668/SKYDROP.FW.zip)

to thest this setup:

1. update to this firmware (save your CFG.EE, so you will have your settings, when you switch back to stable release)
2. Unpair any previous connection with the skydrop
3. delete debug.log from sd card
4. enable debug.log in settings.

Then try it with the XCSoar using classic (SPP):

1. pair with the tablet using android settings
2. go to xcsoar devices and set up device
3. look into device/monitor 

Also you can try it with BLE

1. no pairing!
2. go to xcsoar, set up device
3. use bluetooth LE as port and select SkyDrop BLE
4. look into device/monitor

then attach the debug.log files here (zip them or rename them to txt)

Note: No data in xcsoar can means that you have incorrect protocol set on SkyDrop in Bluetooth/Protocol. I suggest to use Digifly protocol (LK8EX1 is default)

**Comment by piman-34** [comment]
Thanks for a rapid response!I can successfully pair with the SPP profile and then in XCsoar select Skydrop SPP and Digifly and it works :-)  Havent managed to get BLE working yet, will continue in the morning. Also it has broken my original KOBO connection (I guess the key has changed - so will have to reprogram the adaptor).  I have attached a couple of logs......
Will update as I do further tests......
thanks again.
[DEBUG2.txt](https://github.com/fhorinek/SkyDrop/files/409824/DEBUG2.txt)
[DEBUG1.txt](https://github.com/fhorinek/SkyDrop/files/409825/DEBUG1.txt)

**Comment by fhorinek** [comment]
The skydrop is only able to hold one connection key for now.
you did manage to make BLE connection and send data via BLE according to the DEBUG1.txt

**Comment by fhorinek** [comment]
also please disable "record screen" in debug you do not need it and it only drain more power from the device :)

**Comment by piman-34** [comment]
Thanks for the feedback.

I have done some more playing today and yes it does work with BLE, but I find that xcsoar (this is probably not your problem) reports no data when I use Digifly - although I can see the data scrolling through and only gets baro info when I set it to Bluefly. So currently works fine on SPP and Digifly but as you say I lose the connection for my Kobo once I use it on the Android.

I did once get the Kobo to reconnect after working with the android, I will need to look at the setup of the module to maybe retry pairing on each connection as otherwise I have to unsolder it from the Kobo to reprogram it, as I havent found a way to drive the serial port from a telnet connection to the Kobo.

Thanks again for the great support and rapid responses.  I will try my next flight witht he android table to see how I get on with the display etc. and how the batteries all hold up.

**Comment by fhorinek** [comment]
This modification to the firmware will be available in next firmware update. I am closing the issue

**Comment by tchewitt** [comment]
Success for me via SPP. Partial success via BLE as above.

BLE connects automatically from within xcsoar when the BLE device is selected. I see messy data in monitor and "no data" reported in the devices menu. I didn't have GPS fix but can confirm it was not sending baro or vario (which worked on my phone without GPS).

SPP was strange to set up but now works. The device wasn't available in android the first few times. Inside xcsoar I had to select Port:Skydrop SPP / Driver:Digifly instead of Port:Bluetooth server / Driver:Digifly. I have never seen this option before. I see proper data in monitor and xcsoar is receiving baro and vario.

The strange bit is that the android bluetooth menu lists variable devices each time I start it. Always Skydrop BLE with no logo; sometimes a mac address which cannot be paired; sometimes Skydrop SPP with joystick logo and no mac address (assumed related). This option didn't appear the first few times I tried but was the one which succeeded.

Great support and fantastic device standalone. Looking forward to flying with it properly integrated, and to future updates
